### PR TITLE
[AMDGPU][Attributor] Add a pass parameter `closed-world` for AMDGPUAttributor pass

### PR DESCRIPTION
**Pull request by shiltian**

### CODE REVIEW

**Comment by shiltian** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101760?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#101760** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/101760?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#100953** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100953?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#100954](https://github.com/llvm/llvm-project/pull/100954) <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100954?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#100952** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/100952?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shiltian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp [hunk 1]
```diff
@@ -661,6 +662,24 @@ Error AMDGPUTargetMachine::buildCodeGenPipeline(
...
+parseAMDGPUAttributorPassOptions(StringRef Params) {
+  AMDGPUAttributorOptions Result;
+  while (!Params.empty()) {
+    StringRef ParamName;
+    std::tie(ParamName, Params) = Params.split(';');
+    if (ParamName == "closed-world") {
+      Result.IsClosedWorld = true;
+    } else {
+      return make_error<StringError>(
+          formatv("invalid AMDGPUAttributor pass parameter '{0}' ", ParamName)
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp:675]: Add a test for the error case

**Review by arsenm — APPROVED** [review]

**Comment by shiltian** [comment]
This patch will be rebased once https://github.com/llvm/llvm-project/pull/102086 is landed.

**Comment by shiltian** [comment]
The error has been fixed in 23c8128d1ec8030f1a717fd9dc391456b9171e45. Sorry for the noise.

**Pull request merged**